packages/ckeditor5-dev-release-tools/lib/tasks/commitandtag.js

```diff
@@ -39,6 +51,34 @@ export default async function commitAndTag( { version, files, cwd = process.cwd(
 		return;
 	}
 
-	await git.commit( `Release: v${ version }.${ skipCi ? ' [skip ci]' : '' }`, filePathsToAdd );
-	await git.addTag( `v${ version }` );
+	if ( dryRun ) {
+		await executePreCommitHook( { filePathsToAdd, preCommitCommand, cwd } );
+		listrTask.output = `[dry run] Creating git tag v${ tagForVersion }.`;
+	} else {
+		await git.commit( `Release: v${ version }.${ skipCi ? ' [skip ci]' : '' }`, filePathsToAdd );
+		await git.addTag( `v${ version }` );
+	}
+}
+
+/**
+ * Function that imitates running dry run for the `git commit` command.
+ *
+ * @param {object} options
+ * @param {Array.<string>} options.filePathsToAdd File paths to run the dry run validations for.
+ * @param {string} options.preCommitCommand Command to run before creating a commit for the dry run mode.
+ * @param {string} options.cwd Current working directory from which all paths will be resolved.
+ * @returns {Promise<void>}
+ */
+async function executePreCommitHook( { filePathsToAdd, preCommitCommand, cwd } ) {
+	const normalizedCwd = toUnix( cwd );
+	const git = simpleGit( { baseDir: normalizedCwd } );
+
+	try {
+		await git.add( filePathsToAdd );
+		await tools.shExec( preCommitCommand, { cwd, verbosity: 'silent', async: true } );
+	} catch ( e ) {
+		throw e.message;
+	} finally {
+		await git.reset( filePathsToAdd );
+	}
```

Comment:
Two things to consider:

- Instead of executing `yarn lint-staged` command, maybe we could import the `lint-staged` package and call it in Node? Example: https://github.com/lint-staged/lint-staged?tab=readme-ov-file#can-i-use-lint-staged-via-node.
- There is `--diff` option to allow overriding how `lint-staged` gets changed files. If we could call `lint-staged` without adding the files to staging, we could get rid of `git reset` later. See https://github.com/lint-staged/lint-staged?tab=readme-ov-file#command-line-flags.

Reply:
Therefore, we can avoid changes in `ckeditor5-dev`. Instead, we will modify the prepare-release scripts only.

Reply:
Long story short, the `lint-staged` package always adds files to staging, so we cannot avoid executing the `git reset` command.

Let's improve our flow since we cannot find a proper flow using the existing packages, and I'd like to avoid using more than necessary.

When executing `commitAndTag( { dryRun: true } )`, let's do the following:

* Store the latest SHA-1 commit.
* Make a new commit.
* Ensure the tag is not created.
* Execute `git reset --soft {SHA-1}` to remove the committed changes (a revert without a trace in the history).
* Store again the latest SHA-1 commit. Compare both values. If they aren't equal, throw an error. It means that the `dryRun=true` mode is messed up.

Reply:
According to https://stackoverflow.com/a/24569155/3388924, perhaps we should use the `mixed` mode.